io_scene_nif/modules/property/texture/texture_import.py

```diff
@@ -40,364 +40,154 @@
-            if texslot:
-                texslot.offset.x = shader_property.uv_offset.u
-                texslot.offset.y = shader_property.uv_offset.v
-
-    def import_uv_scale(self, b_mat, shader_property):
-        for texslot in b_mat.texture_slots:
-            if texslot:
-                texslot.scale.x = shader_property.uv_scale.u
-                texslot.scale.y = shader_property.uv_scale.v
-
-    def import_texture_game_properties(self, b_mat, shader_property):
-        for texslot in b_mat.texture_slots:
-            if texslot:
-                texslot.texture.image.use_animation = True
-                texslot.texture.image.fps = shader_property.controller.frequency
-                texslot.texture.image.frame_start = shader_property.controller.start_time
-                texslot.texture.image.frame_end = shader_property.controller.stop_time
+        self.texture_loader = TextureLoader()
 
     def create_texture_slot(self, b_mat, image_texture):
         b_mat_texslot = b_mat.texture_slots.add()
         try:
-            b_mat_texslot.texture = self.textureloader.import_texture_source(image_texture.source)
+            b_mat_texslot.texture = self.texture_loader.import_texture_source(image_texture.source)
         except:
-            b_mat_texslot.texture = self.textureloader.import_texture_source(image_texture)
+            b_mat_texslot.texture = self.texture_loader.import_texture_source(image_texture)
         b_mat_texslot.use = True
 
         # Influence mapping
 
         # Mapping
         b_mat_texslot.texture_coords = 'UV'
-        try:
-            b_mat_texslot.uv_layer = self.get_uv_layer_name(image_texture.uv_set)
-        except:
-            b_mat_texslot.uv_layer = self.get_uv_layer_name(0)
-
+        b_mat_texslot.uv_layer = Vertex.get_uv_layer_name(image_texture.uv_set)
         return b_mat_texslot
 
-    def import_image_texture(self, b_mat, n_texture_desc):
+    def update_diffuse_slot(self, b_mat_texslot):
+        # Influence mapping
+        b_mat_texslot.texture.use_alpha = False
 
-        image_texture = n_texture_desc
+        # Influence
+        # TODO [property][texture][flag][alpha] Figure out if this texture has alpha
+        # if self.nif_import.ni_alpha_prop:
+        #     b_mat_texslot.use_map_alpha = True
 
-        b_mat_texslot = self.create_texture_slot(b_mat, image_texture)
+        b_mat_texslot.use_map_color_diffuse = True
 
+    def update_bump_slot(self, b_mat_texslot):
         # Influence mapping
-        if self.has_bumptex:
-            b_mat_texslot.texture.use_normal_map = False  # causes artifacts otherwise.
-        if self.has_normaltex:
-            b_mat_texslot.texture.use_normal_map = True  # causes artifacts otherwise.
-        if self.has_glowtex or self.has_glosstex or self.has_decaltex:
-            b_mat_texslot.texture.use_alpha = False
+        b_mat_texslot.texture.use_normal_map = False  # causes artifacts otherwise.
```

Comment:
Needed to still have two different set of fields so could distinguish between the two, but weird enough the settings needed to be this way as it would not render correctly in the view port.
That said needs testing with some sample files as I have no real nifs, just the ones from the test framework.

Reply:
Actually think this is correct? - `use_normal_map = False`